src/app/policies/views/PolicyListView.vue

```diff
@@ -27,158 +27,205 @@
+                  <template
+                    #loadable="{ data }"
                   >
-                    <AppCollection
-                      class="policy-collection"
-                      data-testid="policy-collection"
-                      :empty-state-message="t('common.emptyState.message', { type: `${policyType.name} policies` })"
-                      :empty-state-cta-to="t('policies.href.docs', { name: policyType.name })"
-                      :empty-state-cta-text="t('common.documentation')"
-                      :headers="[
-                        { label: 'Name', key: 'name' },
-                        { label: 'Namespace', key: 'namespace' },
-                        ...(can('use zones') && policyType.isTargetRefBased ? [{ label: 'Zone', key: 'zone' }] : []),
-                        ...(policyType.isTargetRefBased ? [{ label: 'Target ref', key: 'targetRef' }] : []),
-                        { label: 'Details', key: 'details', hideLabel: true },
-                      ]"
-                      :page-number="route.params.page"
-                      :page-size="route.params.size"
-                      :total="data?.total"
-                      :items="data?.items"
-                      :is-selected-row="(row) => row.id === route.params.policy"
-                      @change="route.update"
+                    <search
+                      v-if="(data?.items ?? { length: 0 }).length > 0 || (route.params.s.length > 0)"
                     >
-                      <template #name="{ row }">
-                        <XAction
-                          :to="{
-                            name: 'policy-summary-view',
-                            params: {
-                              mesh: row.mesh,
-                              policyPath: policyType.path,
-                              policy: row.id,
-                            },
-                            query: {
-                              page: route.params.page,
-                              size: route.params.size,
-                            },
-                          }"
-                        >
-                          {{ row.name }}
-                        </XAction>
-                      </template>
-                      <template #namespace="{ row: item }">
-                        {{ item.namespace.length > 0 ? item.namespace : t('common.detail.none') }}
+                      <form
+                        @submit.prevent
+                      >
+                        <XInput
+                          placeholder="Filter by name..."
+                          type="search"
+                          appearance="search"
+                          :value="route.params.s"
+                          :debounce="1000"
+                          @change="(e) => route.update({
+                            s: e,
+                          })"
+                        />
+                      </form>
+                    </search>
```

Comment:
This change above here in this file is the real change, the rest is mainly a result of re-nesting.